### ISSUE TITLE
Tree Diagram 1-to-many visualization

### ISSUE THREAD

**Issue by getmykhan** [d3blocks/d3blocks#48]
What are your thoughts on having a simplified version of the tree visual to show dataframe/data structure that flows from 1 column to another. 

Consider the table below:

<img width="1191" alt="image" src="https://github.com/d3blocks/d3blocks/assets/15202558/75f6505d-4b13-46b9-ab4b-fa4cb73a030c">

Ideally the tree should be able to link up:
0-1-2-3(0 idx) ; then
0-1-2-3 (1 idx)

Basically it should all branch out from column 0: Which is a unique value; normalized to show consistency

How best to facet this data to achieve the above ^. Because under the hood your code seems to optimize and remove redundancy or duplicates ? So the resulting graph is not like 

0-1-2-3 (n-idx); it could be something which is 2-3-(4 idx)

More context: I did a network representation of this, basically I would want the same representation branching from the largest node to the smallest, but in the form of a tree diagram

<img width="1115" alt="image" src="https://github.com/d3blocks/d3blocks/assets/15202558/8afbbce4-608f-471f-8830-fad209b7a483">

**Comment by getmykhan** [comment]
This probably could be a helpful warning: 

> [d3blocks] >WARNING> The dataframe seems to be circular which can not be handled by this chart!

**Comment by getmykhan** [comment]
@erdogant

**Comment by erdogant** [comment]
Hi, I am not sure whether I understand the issue. Have you tried to restructure your dataset in source-target?

```
source    |  target
# column 1 | column 2
...             | ....

# column 2 | column 3
...             | ....

# column 3 | column 4
...             | ....

```

**Comment by getmykhan** [comment]
@dpttw Sounds similar - do you have a working example. I feel this should a core functionality of the system with large-scale data

**Comment by getmykhan** [comment]
@erdogant I can give you a file "Source" -> "Target" mapping if you want to reproduce it

**Comment by dpttw** [comment]
> @dpttw Sounds similar - do you have a working example. I feel this should a core functionality of the system with large-scale data

Yes, in the energy example, the children of solid are not showing.  My d3graph                  is 2.5.0.
>>> # Load d3blocks
>>> from d3blocks import D3Blocks
>>> #
>>> # Initialize
>>> d3 = D3Blocks()
>>> #
>>> # Load example data
>>> df = d3.import_example('energy')
>>> df.to_csv("sample.csv")
>>> #
>>> # Plot
>>> fileName = os.path.join(os.getcwd(),"sample.html")
>>> d3.tree(df,filepath=fileName)
>>> #

**Comment by getmykhan** [comment]
@erdogant Attaching sample data for you to recreate.

There are two links:
1. The base source file: [sample.csv](https://github.com/user-attachments/files/17247202/sample.csv)
LxJQnhK
2. The mutated source -> target mapping as the library loads it: [sample1.csv](https://github.com/user-attachments/files/17247212/sample1.csv)


Can you run these and share what you are seeing?

Y

**Comment by getmykhan** [comment]
<img width="825" alt="image" src="https://github.com/user-attachments/assets/e8137437-c108-426b-b0cd-e2f3b64ebb2e">

This is what I get
